swupd/create_manifests.go

```diff
@@ -52,18 +52,23 @@ func getOldManifest(path string) (*Manifest, error) {
 	return ParseManifestFile(path)
 }
 
-func initBundles(ui UpdateInfo, c config) ([]*Manifest, error) {
-	var err error
+func initBundles(ui UpdateInfo, c config, numWorkers int) ([]*Manifest, error) {
 	var wg sync.WaitGroup
-	workers := len(ui.bundles)
-	wg.Add(workers)
+	if numWorkers > len(ui.bundles) {
+		numWorkers = len(ui.bundles)
+	}
+	wg.Add(numWorkers)
+
 	bundleChan := make(chan string)
-	errorChan := make(chan error, workers)
+	errorChan := make(chan error, numWorkers)
 	mux := &sync.Mutex{}
 	tmpManifests := []*Manifest{}
+
```

Comment:
just adding newlines?

Reply:
It looked a bit messy, I added new lines for clarity